gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/DatasetInfo.java

```diff
@@ -0,0 +1,409 @@
+    description = builder.description;
+    etag = builder.etag;
+    friendlyName = builder.friendlyName;
+    id = builder.id;
+    lastModified = builder.lastModified;
+    location = builder.location;
+    selfLink = builder.selfLink;
+  }
+
+  /**
+   * Returns the dataset identity.
+   */
+  public DatasetId datasetId() {
+    return datasetId;
+  }
+
+  /**
+   * Returns the dataset's access control configuration.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/access-control">Access Control</a>
+   */
+  public List<Acl> acl() {
+    return acl;
+  }
+
+  /**
+   * Returns the time when this dataset was created, in milliseconds since the epoch.
+   */
+  public Long creationTime() {
+    return creationTime;
+  }
+
+  /**
+   * Returns the default lifetime of all tables in the dataset, in milliseconds. Once this property
+   * is set, all newly-created tables in the dataset will have an expirationTime property set to the
+   * creation time plus the value in this property, and changing the value will only affect new
+   * tables, not existing ones. When the expirationTime for a given table is reached, that table
+   * will be deleted automatically. If a table's expirationTime is modified or removed before the
+   * table expires, or if you provide an explicit expirationTime when creating a table, that value
+   * takes precedence over the default expiration time indicated by this property.
+   */
+  public Long defaultTableLifetime() {
+    return defaultTableLifetime;
+  }
+
+  /**
+   * Returns a user-friendly description for the dataset.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns the hash of the dataset resource.
+   */
+  public String etag() {
+    return etag;
+  }
+
+  /**
+   * Returns a user-friendly name for the dataset.
+   */
+  public String friendlyName() {
+    return friendlyName;
+  }
+
+  /**
+   * Returns an opaque id for the dataset.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the time when this dataset or any of its tables was last modified, in milliseconds
+   * since the epoch.
+   */
+  public Long lastModified() {
+    return lastModified;
+  }
+
+  /**
+   * Returns the geographic location where the dataset should reside.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/docs/managing_jobs_datasets_projects#dataset-location">
+   *     Dataset Location</a>
+   */
+  public String location() {
+    return location;
+  }
+
+  /**
+   * Returns an URL that can be used to access the resource again. The returned URL can be used for
+   * get or update requests.
+   */
+  public String selfLink() {
+    return selfLink;
+  }
+
+  /**
+   * Returns a builder for the {@code DatasetInfo} object.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("datasetId", datasetId)
```

Comment:
No particular reason, I selected "meaningful" fields (as we do in [BlobInfo.toString](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobInfo.java#L533)). But I guess I can simply add all fields.